CAP/gap/TerminalCategory.gi

```diff
@@ -63,106 +422,52 @@ end );
+                                           object,
+                                           object,
+                                           IsOne, true );
     
     SetIsWellDefined( morphism, true );
     
     return morphism;
     
 end );
 
-################################
-##
-## Category functions
-##
-################################
-
-##
-BindGlobal( "INSTALL_TERMINAL_CATEGORY_FUNCTIONS",
-            
-  function( )
-    local obj_function_list, obj_func, morphism_function_list, morphism_function, i;
-    
-    obj_function_list := [ AddZeroObject,
-                           AddKernelObject,
-                           AddCokernelObject,
-                           AddDirectProduct ];
-    
-    obj_func := function( arg ) return UniqueObject( CAP_INTERNAL_TERMINAL_CATEGORY ); end;
-    
-    for i in obj_function_list do
-        
-        i( CAP_INTERNAL_TERMINAL_CATEGORY, obj_func );
-        
-    od;
-    
-    morphism_function_list := [ AddIdentityMorphism,
-                                AddPreCompose,
-                                AddLiftAlongMonomorphism,
-                                AddColiftAlongEpimorphism,
-                                AddInverseForMorphisms,
-                                AddKernelEmbedding,
-                                AddKernelEmbeddingWithGivenKernelObject,
-                                AddKernelLiftWithGivenKernelObject,
-                                AddCokernelProjection,
-                                AddCokernelProjectionWithGivenCokernelObject,
-                                AddCokernelColift,
-                                AddCokernelColiftWithGivenCokernelObject,
-                                AddProjectionInFactorOfDirectProduct,
-                                AddProjectionInFactorOfDirectProductWithGivenDirectProduct,
-                                AddUniversalMorphismIntoDirectProduct,
-                                AddUniversalMorphismIntoDirectProductWithGivenDirectProduct ];
-    
-    morphism_function := function( arg ) return UniqueMorphism( CAP_INTERNAL_TERMINAL_CATEGORY ); end;
-    
-    for i in morphism_function_list do
-        
-        i( CAP_INTERNAL_TERMINAL_CATEGORY, morphism_function );
-        
-    od;
-    
-end );
-
-INSTALL_TERMINAL_CATEGORY_FUNCTIONS( );
-
-################################
-##
-## Functor constructors
-##
-################################
+##################################
+#
+# Functors
+#
+##################################
 
 ##
 InstallMethod( FunctorFromTerminalCategory,
                [ IsCapCategoryObject ],
-               
+  
   function( object )
     local functor;
     
-    functor := CapFunctor( Concatenation( "InjectionInto", Name( CapCategory( object ) ) ), CAP_INTERNAL_TERMINAL_CATEGORY, CapCategory( object ) );
+    functor := CapFunctor( Concatenation( "InjectionInto", Name( CapCategory( object ) ) ), TerminalCategory( ), CapCategory( object ) );
```

Comment:
Just keep the old version: I don't think we always want to construct a new TerminalCategory here.